### ISSUE TITLE
watchexec is unreliably when watching Windows filesystem within WSL

### ISSUE THREAD

**Issue by cswinter** [watchexec/watchexec#153]
I use watchexec (via cargo watch) under Windows Subsystem for Linux (WSL) to watch Rust code that is stored on a Windows filesystem. A lot of the time this works, but usually cargo watch will eventually fail to respond to any file modification events. Killing/restarting cargo watch usually fixes it (for a while). Presumably this is a just an inherent technical limitation of or bug in WSL, but it would be great if a fix or workaround were possible in watchexec.

Let me know if there is any diagnostics I can collect to help debug this issue.

**Comment by passcod** [comment]
You can always use polling mode.

A verbose log wouldn't be amiss, as well as windows build number, if it's wsl 1 or 2, etc?

As another potential workaround, you could run a watcher on the windows side and communicate somehow (possibly through touching a file to trigger another watcher on the wsl side! if that's reliable) to use the windows APIs instead of the translated linux ones.

**Comment by cswinter** [comment]
Thanks for the help, I've started using the `--poll` option and so far it's worked great, guess I should have RTFM 😄 

I've tried to collect some more info by running with `--debug` but so far I haven't seen any differences in the output from when it's working vs when not. I _think_ I'm still on WSL 1 unless it upgrades automatically, `uname -a` gives me this:

`Linux DESKTOP-EBK0B57 4.4.0-18362-Microsoft #836-Microsoft Mon May 05 16:04:00 PST 2020 x86_64 x86_64 x86_64 GNU/Linux`

I have a good workaround now and I'm pretty happy to just chalk it up to a WSL bug at this point (I've seen similar things with e.g. `tail` hanging after a file gets replaced) so feel free to close the issue.
